### DISCUSSION TITLE
Getting field name out of Prisma validation error

### DISCUSSION THREAD

**Discussion by shchepinsky** [prisma/prisma#7267] [Q&A] [answered]
I'm trying to get field name from Prisma validation error to return it for client. There are two king of errors thrown on bad requests:

For example, if the field is missing, it throws https://www.prisma.io/docs/reference/api-reference/error-reference/#prismaclientvalidationerror which looks like this:

```
{
  "error": "\nInvalid `prisma.employer.create()` invocation:\n\n{\n  data: {\n+   label: String,\n?   id?: String,\n?   contracts?: {\n?     create?: EmployeeContractsCreateWithoutEmployerInput | EmployeeContractsCreateWithoutEmployerInput | EmployeeContractsUncheckedCreateWithoutEmployerInput | EmployeeContractsUncheckedCreateWithoutEmployerInput,\n?     connectOrCreate?: EmployeeContractsCreateOrConnectWithoutEmployerInput | EmployeeContractsCreateOrConnectWithoutEmployerInput,\n?     connect?: EmployeeContractsWhereUniqueInput | EmployeeContractsWhereUniqueInput\n?   }\n  }\n}\n\nArgument label for data.label is missing.\n\nNote: Lines with + are required, lines with ? are optional.\n"
}
```

The only barely useful data for a client is at the ending part: `Argument label for data.label is missing`.

At the same time, if constraint are not met, it throws a different error https://www.prisma.io/docs/reference/api-reference/error-reference/#prismaclientknownrequesterror

```
{
  "code": "P2002",
  "clientVersion": "2.21.2",
  "meta": {
    "target": [
      "label"
    ]
  }
}
```

 ...where meta.target contains offending fields.

What is the proper way of getting field names from Prisma errors in the first case?

It would nice if Prisma Error objects had standardised per-field error map like `{ label: "is missing" }`, `{ label: "unique value required" }`.

**Comment by janpio** [comment, marked as answer]
That looks like a feature request to me - maybe recreate it as that and we can try to improve our error messages in an validation error.

**Reply by ryands17** [reply]
@shchepinsky 👋 

Yes it would be great if you could create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=).

**Reply by shchepinsky** [reply]
@ryands17 @janpio thanks guys. Sure, here it is: https://github.com/prisma/prisma/issues/7285
